lib/lightning/accounts/user.ex

```diff
@@ -164,14 +164,74 @@ defmodule Lightning.Accounts.User do
     |> put_change(:role, :superuser)
   end
 
-  def validate_email(changeset) do
+  @doc """
+  Validates that a user has access to all projects associated with a credential.
+  Returns a changeset with any project access errors.
+  """
+  def validate_project_access(changeset, credential) do
+    case get_field(changeset, :email) do
+      nil ->
+        changeset
+
+      email ->
+        case Lightning.Repo.exists?(User |> where(email: ^email)) do
+          false ->
+            changeset
+
+          true ->
+            user = Lightning.Accounts.get_user_by_email(email)
```

Comment:
Can be done in a single query and check for `nil`

Reply:
This is fixed, thanks for the catch